src/TokenManagers/XboxTokenManager.js

```diff
@@ -138,6 +138,39 @@ class XboxTokenManager {
     }
   }
 
+  async doSisuAuth (accessToken, deviceToken, options) {
+    const payload = {
+      AccessToken: 't=' + accessToken,
+      AppId: options.authTitle,
+      DeviceToken: deviceToken,
+      Sandbox: 'RETAIL',
+      UseModernGamertag: true,
+      SiteName: 'user.auth.xboxlive.com',
+      RelyingParty: options.relyingParty || 'http://xboxlive.com/',
+      ProofKey: this.jwk
+    }
+
+    const body = JSON.stringify(payload)
+
+    const signature = this.sign(Endpoints.SisuAuthorize, '', body).toString('base64')
+
+    const headers = { Signature: signature }
+
+    const ret = await fetch(Endpoints.SisuAuthorize, { method: 'post', headers, body }).then(checkStatus)
+    debug('Sisu Auth Response', ret)
+
+    const xsts = {
```

Comment:
Can this fail (that would result in missing JSON fields)? If so improved error handling here would be good, like is handled below in getXSTSTokenWithTitle.

Reply:
Added. Sisu has the XErr code in the response headers so didn't notice it originally.